File: sample-videochat-webrtc/src/main/java/com/quickblox/sample/groupchatwebrtc/fragments/BaseFragment.java
package com.quickblox.sample.groupchatwebrtc.fragments;

import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.quickblox.sample.groupchatwebrtc.utils.WebRtcSessionManager;
import com.quickblox.videochat.webrtc.QBRTCSession;

/**
 * Created by tereha on 24.05.16.
 */
public abstract class BaseFragment extends Fragment {

    private WebRtcSessionManager sessionManager;

    private QBRTCSession currentSession;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(getCurrentView(), container, false);

        initFields();
        initUi(view);

        return view;
    }


    private void initFields() {
        sessionManager = WebRtcSessionManager.getInstance(getActivity());
        currentSession = sessionManager.getCurrentSession();
    }

    protected QBRTCSession getCurrentSession() {
        return currentSession;
    }

    abstract int getCurrentView();
    protected abstract void initUi(View view);


    @Override
    public void onStart() {
        super.onStart();
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();
    }

    @Override
    public void onStop() {
        super.onStop();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }
}

File: sample-conference-kotlin/app/src/main/java/com/quickblox/sample/conference/kotlin/presentation/screens/base/BaseFragment.kt
package com.quickblox.sample.conference.kotlin.presentation.screens.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.viewbinding.ViewBinding

/*
 * Created by Injoit in 2021-09-30.
 * Copyright © 2021 Quickblox. All rights reserved.
 */
abstract class BaseFragment<VB : ViewBinding> : Fragment() {
    var binding: VB? = null
        private set

    abstract fun getViewBinding(inflater: LayoutInflater, container: ViewGroup?): VB

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = getViewBinding(inflater, container)
        return binding?.root
    }

    override fun onDestroyView() {
        binding = null
        super.onDestroyView()
    }
}